pkg/ingress/status.go

```diff
@@ -23,30 +23,61 @@ import (
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/apache/apisix-ingress-controller/pkg/kube"
+	configv1 "github.com/apache/apisix-ingress-controller/pkg/kube/apisix/apis/config/v1"
 	configv2alpha1 "github.com/apache/apisix-ingress-controller/pkg/kube/apisix/apis/config/v2alpha1"
+	"github.com/apache/apisix-ingress-controller/pkg/log"
 )
 
 const (
 	_conditionType        = "ResourcesAvailable"
 	_commonSuccessMessage = "Sync Successfully"
 )
 
-// recordRouteStatus record ApisixRoute v2alpha1 status
-func recordRouteStatus(ar *configv2alpha1.ApisixRoute, reason, message string, status v1.ConditionStatus) {
+// recordStatus record resources status
+func recordStatus(at interface{}, reason string, err error, status v1.ConditionStatus) {
 	// build condition
+	message := _commonSuccessMessage
+	if err != nil {
+		message = err.Error()
+	}
 	condition := metav1.Condition{
 		Type:    _conditionType,
 		Reason:  reason,
 		Status:  status,
 		Message: message,
 	}
 
-	// set to status
-	if ar.Status.Conditions == nil {
-		conditions := make([]metav1.Condition, 0)
-		ar.Status.Conditions = &conditions
+	switch v := at.(type) {
+	case *configv1.ApisixTls:
+		// set to status
+		if v.Status.Conditions == nil {
+			conditions := make([]metav1.Condition, 0)
+			v.Status.Conditions = &conditions
+		}
+		meta.SetStatusCondition(v.Status.Conditions, condition)
+		_, _ = kube.GetApisixClient().ApisixV1().ApisixTlses(v.Namespace).
```

Comment:
Why ignore both the return values?

Reply:
I thought about that error message should be useful.
Added the error logs.

Reply:
Got it.